app/services/container_dashboard_service.rb

```diff
@@ -133,6 +134,7 @@ def heatmaps
       node_memory_usage = []
 
       metrics.each do |m|
+        next if m.resource.nil? # Metrics are purged asynchronously and might be missing their node
```

Comment:
can you just look at `m.resource_id` instead?
If this is purged, then what exactly happens here?
if the resource is purged, should this record be invalid?